node_modules/oae-core/createlink/js/createlink.js

```diff
@@ -309,8 +309,13 @@ define(['jquery', 'oae.core', 'jquery.jeditable'], function($, oae) {
 
                 // Update the members of the added links
                 $.each(addedLinks, function(index, link) {
-                    link.viewers = _.without(data.shared, oae.data.me.id);
-                    link.folders = data.folders;
+                    link.viewers = _.chain(data.selectedPrincipalItems)
+                        .filter(function(selectedPrincipalItem){
```

Comment:
Same here about the filter. However, if we just pass in the `shareId` array and the folder id array, this can easily be simplified again.